cirrus/google_cloud/iam.py

```diff
@@ -0,0 +1,166 @@
+        """
+        self.role = role
+        self.members = members
+
+    def __repr__(self):
+        """
+        Return representation of object
+
+        Returns:
+            str: Representation of the Binding which can be POSTed to Google's API
+        """
+        output_dict = dict()
+        output_dict["role"] = str(self.role)
+        output_dict["members"] = [str(member) for member in self.members]
+        return str(output_dict)
+
+
+class GooglePolicyMember(object):
+    """
+    A Member for a Google Policy
+    """
+
+    SERVICE_ACCOUNT = "serviceAccount"
+    USER = "user"
+    GROUP = "group"
+    DOMAIN = "domain"
+
+    def __init__(self, member_type, email_id=""):
+        """
+        Construct the Member for the Google Policy
+
+        Args:
+            name (str): Description
+            member_type (str): Type of member (see Google's definition below)
+
+            .. code-block:: yaml
+
+            allUsers:
+                - A special identifier that represents anyone who is on the internet;
+                  with or without a Google account.
+            allAuthenticatedUsers:
+                - A special identifier that represents anyone who is authenticated
+                  with a Google account or a service account.
+            user (requires email_id):
+                - An email address that represents a specific Google account.
+                  For example, alice@gmail.com or joe@example.com.
+            serviceAccount (requires email_id):
+                - An email address that represents a service account.
+                  For example, my-other-app@appspot.gserviceaccount.com.
+            group (requires email_id):
+                - An email address that represents a Google group.
+                  For example, admins@example.com.
+            domain (requires domain as email_id):
+                - A Google Apps domain name that represents all the users of
+                  that domain. For example, google.com or example.com.
+
+        """
+        self.member_type = member_type
+        self.email_id = email_id
+
+    def __repr__(self):
+        """
+        Return representation of object
+
+        Returns:
+            str: Representation of the Member for Google's API
+        """
+        output = "{}:{}".format(self.member_type, self.email_id)
+        return output
+
+
+class GooglePolicyRole(object):
+    """
+    A Role for use in a Google Policy
+    """
+    ROLE_PREFIX = "roles/"
+
+    def __init__(self, name):
+        """
+        Construct the Role
+
+        Args:
+            name (str): The name of the Google role
+        """
+        # If the name provided already starts with the prefix, remove it
+        if name.strip()[:len(GooglePolicyRole.ROLE_PREFIX)] == GooglePolicyRole.ROLE_PREFIX:
+            name = name.strip()[len(GooglePolicyRole.ROLE_PREFIX):]
+
+        self.name = name
+        # TODO check if it's an actual role in Google cloud
```

Comment:
Is this a "future issue" type of TODO?

Reply:
eh, that should proably just get removed. Can't really ping Google's API from this class to see if it's valid, so the issue will get handled elsewhere if it's not a valid role and someone tries to use it